homeassistant/helpers/selector.py

```diff
@@ -38,6 +48,13 @@ class Selector:
     """Base class for selectors."""
 
     CONFIG_SCHEMA: Callable
+    config: ConfigType
+    hass: HomeAssistant
+
+    def __init__(self, hass: HomeAssistant, config: ConfigType) -> None:
```

Comment:
There are 2 types of validation. One that the format is correct and one where it validates that the shape of the data is correct.

We should only validate that the format is correct inside voluptuous.